plugin/src/main/java/org/wso2/carbon/uuf/maven/bean/AppConfig.java

```diff
@@ -356,44 +356,62 @@ public void setSubmenus(List<MenuItem> submenus) {
          */
-        public PatternsConfig getCsrfPatterns() {
-            return csrfPatterns;
+        public List<String> getCsrfIgnoreUris() {
+            return csrfIgnoreUris;
         }
 
         /**
-         * Sets the CSRF URI patterns of this security configuration.
+         * Sets the list of URI's that doesn't require CSRF protection.
          *
-         * @param csrfPatterns CSRF URI patterns to be set
+         * @param csrfIgnoreUris list of URI's that doesn't require CSRF protection.
+         * @throws IllegalArgumentException if URIs in {@code csrfIgnoreUris} is empty or doesnn't start with '/'.
          */
-        public void setCsrfPatterns(PatternsConfig csrfPatterns) {
-            this.csrfPatterns = (csrfPatterns == null) ? new PatternsConfig() : csrfPatterns;
+        public void setCsrfIgnoreUris(List<String> csrfIgnoreUris) {
+            for (String csrfUri : csrfIgnoreUris) {
+                if (csrfUri.isEmpty()) {
+                    throw new IllegalArgumentException("URIs configured in 'csrfIgnoreUris' cannot be empty.");
+                } else if (!csrfUri.startsWith("/")) {
+                    throw new IllegalArgumentException(
+                            "URIs configured in 'csrfIgnoreUris' must start with a '/'. ");
+                }
+            }
+            this.csrfIgnoreUris = (csrfIgnoreUris == null) ? Collections.emptyList() : csrfIgnoreUris;
         }
 
         /**
-         * Returns XSS URI patterns of this security configuration.
+         * Returns the list of URI's that doesn't require XSS protection.
          *
-         * @return XSS URI patterns
+         * @return list of URI's that doesn't require XSS protection.
          */
-        public PatternsConfig getXssPatterns() {
-            return xssPatterns;
+        public List<String> getXssIgnoreUris() {
+            return xssIgnoreUris;
         }
 
         /**
-         * Sets the XSS URI patterns of this security configuration.
+         * Sets the list of URI's that doesn't require XSS protection.
          *
-         * @param xssPatterns XSS URI patterns to be set
+         * @param xssIgnoreUris the list of URI's that doesn't require XSS protection.
+         * @throws IllegalArgumentException if URIs in {@code xssIgnoreUris} is empty or doesn't start with '/'.
          */
-        public void setXssPatterns(PatternsConfig xssPatterns) {
-            this.xssPatterns = (xssPatterns == null) ? new PatternsConfig() : xssPatterns;
+        public void setXssIgnoreUris(List<String> xssIgnoreUris) {
+            for (String xssUri : xssIgnoreUris) {
+                if (xssUri.isEmpty()) {
+                    throw new IllegalArgumentException("URIs configured in 'xssIgnoreUris' cannot be empty.");
+                } else if (!xssUri.startsWith("/")) {
+                    throw new IllegalArgumentException(
+                            "URIs configured in 'xssIgnoreUris' must start with a '/'. ");
```

Comment:
Shall we change this exception message to `"XSS ignore URI in the app's config must start with a '/'. Instead found '" +xssUri.charAt(0) + "' at the beginning of the ignore URI '" + xssUri + "'."`